plugins/funder.c

```diff
@@ -744,85 +749,192 @@ param_policy_mod(struct command *cmd, const char *name,
+			  || channel_fee_basis || channel_base_msat))
+		   || (leases_avail && *leases_avail))
+			policy->ad = liquidity_ad_default(policy);
+		else
+			policy->ad = tal_steal(policy, current_policy->ad);
+
+		if (lease_basis)
+			policy->ad->lease_basis = *lease_basis;
+		if (lease_base_sat)
+			policy->ad->lease_base_sat = *lease_base_sat;
+		if (funding_weight_max)
+			policy->ad->funding_weight_max = *funding_weight_max;
+		if (channel_fee_basis)
+			policy->ad->channel_fee_basis = *channel_fee_basis;
+		if (channel_base_msat)
+			policy->ad->channel_base_msat = *channel_base_msat;
+	}
+
 
 	err = funder_check_policy(policy);
-	if (err)
+	if (err) {
+		tal_free(policy);
 		return command_done_err(cmd, JSONRPC2_INVALID_PARAMS,
 					err, NULL);
+	}
 
+	tal_free(current_policy);
 	current_policy = policy;
-	res = jsonrpc_stream_success(cmd);
-	policy_to_json(res, current_policy);
-	return command_finished(cmd, res);
+
+	/* Update lightningd, also */
+	req = jsonrpc_request_start(cmd->plugin, cmd,
+				    "setliquidityad",
+				    &liquidityad_set,
+				    &forward_error,
+				    current_policy);
+
+	if (current_policy->ad)
+		json_add_liquidity_ad(req->js, current_policy->ad);
+	else
+		json_add_liquidity_ad(req->js, liquidity_ad_empty(tmpctx));
+
+	return send_outreq(cmd->plugin, req);
 }
 
-static const struct plugin_command commands[] = { {
+static const struct plugin_command commands[] = {
+	{
 		"funderupdate",
-		"channels",
-		"Update configuration for dual-funding offer",
-		"Update current funder settings. Modifies how node"
-		" reacts to incoming channel open requests. Responds with list"
+		"liquidity",
+		"Configuration for dual-funding settings.",
+		"Update current settings. Modifies how node reacts to"
+		" incoming channel open requests. Responds with list"
 		" of current configs.",
 		json_funderupdate
-	}
+	},
 };
 
 static const char *init(struct plugin *p, const char *b, const jsmntok_t *t)
 {
 	const char *err;
+	struct json_out *jout;
+	struct liquidity_ad *ad = current_policy->ad;
+	struct amount_sat val;
 
 	list_head_init(&pending_opens);
 
 	err = funder_check_policy(current_policy);
 	if (err)
 		plugin_err(p, "Invalid parameter combination: %s", err);
 
+	if (!ad)
+		ad = liquidity_ad_empty(tmpctx);
+
+	/* Update lightningd with our latest things */
+	jout = json_out_new(NULL);
+	json_out_start(jout, NULL, '{');
+	json_out_addstr(jout, "lease_fee_base_msat",
+			type_to_string(tmpctx, struct amount_sat,
+				       &ad->lease_base_sat));
+	json_out_add(jout, "lease_fee_basis", false,
+		     "%d", ad->lease_basis);
+	json_out_add(jout, "lease_weight_max", false,
+		     "%d", ad->funding_weight_max);
+	json_out_addstr(jout, "lease_channel_fee_base_max_msat",
+			type_to_string(tmpctx, struct amount_msat,
+				       &ad->channel_base_msat));
+	json_out_add(jout, "lease_channel_fee_basis_max", false,
+		     "%d", ad->channel_fee_basis);
+	json_out_end(jout, '}');
```

Comment:
How is this different from json_add_liquidity_ad ?

Reply:
`json_add_liquididty_ad` operates on a `json_stream` object, this is a raw `json_out`